src/main/java/ArrayQueue.java

```diff
@@ -93,4 +94,27 @@ public String toString() {
         }
         return builder.toString();
     }
+
+    @Override
+    public final boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        ArrayQueue<?> that = (ArrayQueue<?>) o;
+        if (this.size != that.size) return false;
+        for (int i = 0; i < this.size; i++) {
+            if (!this.queue[(this.front + i) % this.queue.length].equals(that.queue[(that.front + i) % that.queue.length])) {
```

Comment:
A private method for getting the next index would clean this up a bit. Can also use throughout the rest of the class.

Reply:
To give an idea of my current brain condition:
I was writing the equals method with `(this.front + i) % this.queue.length` and I thought yo myself "you know, it would be could have something, kinda' like a constant, but it was like a math function instead. Well, _C'est la vie_." Somehow, the thought of writing a... function... didn't cross my mind.